client/state/data-layer/wpcom/account-recovery/lookup/index.js

```diff
@@ -32,12 +35,15 @@ export const validate = ( { primary_email, primary_sms, secondary_email, seconda
 	}
 };
 
-export const handleRequestResetOptions = ( { dispatch }, { userData } ) => (
+export const handleRequestResetOptions = ( { dispatch }, { userData }, next, transit = page ) => (
```

Comment:
> I feel like this redirect should happen as a result of some submitUsername UI action or similar - handleRequestResetOptions doesn't seem to have any relation to the UI but is triggering a route change

That's a great question! Actually, it is also what I'm struggling on. I think the scenario here is: when a user is at the page /account-recovery/ and receive a successful response for requesting the reset options, redirect the user to the next page. 

Normally we could fire the async request from the `onSubmit` handler, and trigger the route change from the `Promise.then()` callback. However, we have moved the actual async data flow to the middleware. Thus, I couldn't find an exact match from the UI side and thus did it here. Maybe I've overlooked something?

> Also, could we use a name like redirect instead of transit?
Sure. I will update it soon :)